StringOperationManager.java

```diff
@@ -0,0 +1,241 @@
+                               break;
+
+                        case 5:
+                        System.out.print(" not available ");
+;
+                                break;
+
+                        case 6:
+                        System.out.println("not available ");
+                        // String[] splitResult = split(UserString);
+                        // for (String word : splitResult) {
+                        //     System.out.println(word);
+                        // }
+                              break;
+
+                        case 7:
+                        System.out.println("not available ");
+                              break;
+
+                        case 8:
+                        System.out.println("Sorted string: " + sort(UserString));
+                              break;
+
+                        case 9:
+                        System.out.print("not available ");
+                        
+                               break;
+
+                        case 10:
+                        System.out.println("Reversed string: " + reverse(UserString ));
+                               break;
+
+                        default: 
+                        System.out.println("Invalid choice.");
+                }
+        
+                
+            }
+        
+    }    
+    
+    public static boolean ValidString(String S){
+        for( int i = 0 ; i < S.length() ; i++){
+          if (Character.isDigit(S.charAt(i))){
+            System.out.println("enter a valid string");
+            return false;  
+           }   
+        }
+          return true;
+    }
+
+
+    public static String AppendString(String first ,String next ){
+         return UserString + " " + next ;
+
+    }
+
+
+    public static int countWords(String str) {
+        int count = 0;
+		boolean choice  = false;
+		int endword = str.length() - 1;
+		for (int i = 0;i < str.length();i++)
+		{
+			if (Character.isLetter(str.charAt(i)) && i != endword)
+					{
+				choice  = true;
+					}
+			else if ( !Character.isLetter(str.charAt(i)) && choice )
+			{
+				count ++;
+				choice  = false;
+			}
+			else if (Character.isLetter(str.charAt(i)) && i == endword)
+			{
+				count ++;
+			}
+		}
+		return count ;
+    }
+
+   
+    public static String replace(String str, char oldSubstring, char newSubstring) {
+     
+        char[] result = new char[str.length()];
+
+		for(int i = 0;i < str.length();i++)
+
+		   {
+			if(str.charAt(i) == oldSubstring){
+			
+				result[i] = newSubstring;
+			}
+			else 
+			{
+				result[i] = str.charAt(i);
+			}
+			}
+		return new String(result);
+    }
+
+
+
+    public static boolean isPalindrome(String in) {
+        if(in == null || in.length() == 0)
+		{
+			return true;
+		}
+
+		for(int i = 0; i <= in.length()/2;i++)
+		{
+			char start = in.charAt(i);
+			char end =in.charAt(in.length() - 1 - i);
+
+		if (start != end)

```

Comment:
proper indentation and code ethics must be followed to increase code readability